src/plugins/SetDocumentationSection/set_documentation_section.ts

```diff
@@ -0,0 +1,55 @@
+import { Application } from "probot";
+import { REPO_HOME_ASSISTANT_IO } from "../../const";
+import { IssueContext } from "../../types";
+import { filterEventNoBot } from "../../util/filter_event_no_bot";
+import { extractDocumentationSectionsLinks } from "../../util/text_parser";
+import { filterEventByRepo } from "../../util/filter_event_repo";
+
+export const NAME = "SetDocumentationSection";
+
+export const initSetDocumentationSection = (app: Application) => {
+  app.on(
+    ["issues.opened"],
+    filterEventNoBot(
+      NAME,
+      filterEventByRepo(
+        NAME,
+        REPO_HOME_ASSISTANT_IO,
+        runSetDocumentationSection
+      )
+    )
+  );
+};
+
+export const runSetDocumentationSection = async (context: IssueContext) => {
+  const foundSections = extractDocumentationSectionsLinks(
+    context.payload.issue.body
+  );
+
+  if (foundSections.includes("integration")) {
+    // Don't do anything for integration sections
+    return;
+  }
+
+  let labels = (await Promise.all(
+    foundSections.map(async (section) => {
+      const exist = await context.github.issues.getLabel(
```

Comment:
Why do we have to check if the label exists? they are created on the fly if I recall correctly from the labelbot.

Reply:
To keep some sort of control on the labels

Reply:
in this case, it is based on user input. We don't want invalid user input to create a bunch of random labels.